set.go

```diff
@@ -0,0 +1,293 @@
+package ksuid
+
+import "encoding/binary"
+
+// CompressedSet is an immutable data type which stores a set of KSUIDs.
+type CompressedSet []byte
+
+// Iter returns an iterator that produces all KSUIDs in the set.
+func (set CompressedSet) Iter() CompressedSetIter {
+	return CompressedSetIter{
+		content: []byte(set),
+	}
+}
+
+// Compress creates and returns a compressed set of KSUIDs from the list given
+// as arguments.
+func Compress(ids ...KSUID) CompressedSet {
+	c := byteLength + (len(ids) / 5)
+	b := make([]byte, 0, c)
+	return AppendCompressed(b, ids...)
+}
+
+// AppendCompressed uses the given byte slice as pre-allocated storage space to
+// build a KSUID set.
+//
+// Note that the set uses a compression technique to store the KSUIDs, so the
+// resuling length is not 20 x len(ids). The rule of thumb here is for the given
+// byte slice to reserve the amount of memory that the application would be OK
+// to waste.
+func AppendCompressed(set []byte, ids ...KSUID) CompressedSet {
```

Comment:
What if the supplied slice is smaller than the amount of memory required to build the compressed set?

Reply:
It's and `append`, it'll get extended as needed.